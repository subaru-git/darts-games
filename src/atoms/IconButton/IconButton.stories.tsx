import React from 'react';
import { ComponentMeta, Story } from '@storybook/react';
import { MdDeleteForever } from 'react-icons/md';
import IconButton from './IconButton';

export default {
  title: 'Atoms/IconButton',
  component: IconButton,
} as ComponentMeta<typeof IconButton>;

const Template: Story<{ colors: string[] }> = (args) => (
  <div className='flex gap-1'>
    {args.colors.map((color) => (
      <IconButton key={color} color={color}>
        <MdDeleteForever size={24} />
      </IconButton>
    ))}
  </div>
);

export const Default = Template.bind({});
Default.args = {
  colors: ['green', 'green-fill', 'blue', 'blue-fill', 'orange', 'orange-fill', 'ghost'],
};
