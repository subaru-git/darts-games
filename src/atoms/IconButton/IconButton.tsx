import React, { ButtonHTMLAttributes, FC } from 'react';

type IconButtonProps = {
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
  disabled?: boolean;
  color?: string;
  children?: React.ReactNode;
};

const IconButton: FC<IconButtonProps & ButtonHTMLAttributes<HTMLButtonElement>> = ({
  onClick,
  children,
  disabled,
  color = 'blue-fill',
  ...props
}) => {
  return (
    <button
      type='button'
      className={`${style[color]} m-auto inline-flex items-center rounded-full p-2.5 text-center text-sm font-medium focus:outline-none focus:ring-4 disabled:cursor-not-allowed`}
      onClick={(e) => onClick && onClick(e)}
      {...props}
      disabled={disabled}
    >
      {children}
    </button>
  );
};

const style: { [key: string]: string } = {
  green: 'text-emerald-600 border-emerald-600 bg-white hover:bg-emerald-50',
  'green-fill':
    'text-white bg-emerald-600 border-emerald-200 hover:bg-emerald-500 disabled:bg-emerald-200',
  blue: 'text-blue-500 border-blue-400 bg-white hover:bg-blue-50',
  'blue-fill': 'text-white bg-blue-500 hover:bg-blue-400 disabled:bg-blue-300',
  orange: 'text-orange-400 border-orange-400 bg-white hover:bg-orange-50',
  'orange-fill': 'text-white bg-orange-500 hover:bg-orange-400  disabled:bg-orange-200',
  ghost: 'text-gray-600 bg-transparent hover:text-gray-400 p-0 focus:ring-0',
} as const;

export default IconButton;
