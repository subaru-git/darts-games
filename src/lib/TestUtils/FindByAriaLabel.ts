export const findByAriaLabel = (elements: HTMLElement[], label: string) => {
  return (
    elements.find((button) => button.attributes.getNamedItem('aria-label')?.value === label) ??
    new HTMLElement()
  );
};
