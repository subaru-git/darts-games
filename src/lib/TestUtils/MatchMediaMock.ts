import mediaQuery from 'css-mediaquery'

export const createMatchMedia =
  (width: number) =>
  (query: string): MediaQueryList => ({
    matches: mediaQuery.match(query, { width }),
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })
