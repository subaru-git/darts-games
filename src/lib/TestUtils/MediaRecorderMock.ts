export const MediaRecorderMock = {
  start: jest.fn(),
  ondataavailable: jest.fn(),
  onerror: jest.fn(),
  state: '',
  stop: jest.fn(),
};
