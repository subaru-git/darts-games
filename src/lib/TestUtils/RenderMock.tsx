import { ReactNode } from 'react';
import { ChakraProvider } from '@chakra-ui/react';
import { render as defaultRender } from '@testing-library/react';
import Providers from '@/contexts/Providers';

const render = (children: ReactNode | ReactNode[]) =>
  defaultRender(
    <ChakraProvider>
      <Providers>{children}</Providers>
    </ChakraProvider>,
  );
export { render };
