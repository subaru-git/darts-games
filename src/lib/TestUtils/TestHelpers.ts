import { waitFor as originalWaitFor, waitForOptions } from '@testing-library/react';

const defaultTimeout = process.env.WAIT_FOR_DEFAULT_TIMEOUT ? parseInt(process.env.WAIT_FOR_DEFAULT_TIMEOUT, 10) : 5000;

export async function waitFor(callback: () => void,  options?: waitForOptions): Promise<void> {
  return originalWaitFor(callback, { timeout: defaultTimeout, ...options });
}