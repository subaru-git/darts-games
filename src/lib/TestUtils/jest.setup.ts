import mocksdk from 'firebase-mock';
import failOnConsole from 'jest-fail-on-console';
import { createMatchMedia } from '@/lib/TestUtils/MatchMediaMock';
import { MediaRecorderMock } from '@/lib/TestUtils/MediaRecorderMock';

jest.mock('next/router', () => ({ useRouter: jest.fn().mockReturnValue({ locale: 'en' }) }));
jest.mock('react-camera-pro');
jest.mock('@/firebase/client', () => ({
  analytics: mocksdk.analytics,
  db: mocksdk.firestore,
  storage: mocksdk.storage,
  auth: mocksdk.auth,
  functions: mocksdk.functions,
}));
jest.mock('@firebase/auth');
beforeAll(() => {
  window.matchMedia = createMatchMedia(1100);
  window.MediaRecorder = (jest.fn() as any).mockImplementation(() => MediaRecorderMock);
  global.ResizeObserver = require('resize-observer-polyfill');
});

failOnConsole();
