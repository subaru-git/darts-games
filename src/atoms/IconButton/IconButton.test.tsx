import IconButton from './IconButton';
import { render } from '@/lib/TestUtils/RenderMock';

test('should rendering', () => {
  const { container } = render(<IconButton />);
  expect(container).toMatchSnapshot();
});
