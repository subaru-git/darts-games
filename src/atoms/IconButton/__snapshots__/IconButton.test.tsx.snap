// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering 1`] = `
<div>
  <button
    class="m-auto inline-flex items-center rounded-full bg-blue-500 p-2.5 text-center text-sm font-medium text-white hover:bg-blue-400 focus:outline-none focus:ring-4 focus:ring-blue-300 disabled:cursor-not-allowed disabled:bg-blue-300"
    type="button"
  />
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
